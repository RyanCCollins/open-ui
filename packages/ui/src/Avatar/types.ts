import * as React from 'react';
import Avatar from './';

export type ImageSize = 'thumb' | 'small' | 'medium' | 'large';
export interface Props extends React.Props<typeof Avatar> {
  src: string;
  name?: string;
  size?: ImageSize;
}
