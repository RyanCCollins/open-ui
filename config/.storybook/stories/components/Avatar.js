import React from 'react';
import { storiesOf } from '@kadira/storybook';
import { Avatar } from 'ui';
import { 
  withKnobs, 
  text, 
  select 
} from '@kadira/storybook-addon-knobs';

const contributor = {
  name: 'Ryan Collins',
  src: 'https://avatars0.githubusercontent.com/u/13810084?v=3&s=460',
};

const sizeOptions = {
  thumb: 'thumb',
  small: 'small',
  medium: 'medium',
  large: 'large',
}

const stories = storiesOf('Avatar', module);
stories.addDecorator(withKnobs);

stories
  .addWithInfo(
    'default avatar',
    `
      Default Avatar
    `,
     () => (
      <Avatar />
    ),
    {
      inline: true,
      propTables: false,
    }
  )
  .addWithInfo(
    'with size prop',
    `
      Properties:

      \`size: thumb | small | medium | large \` 

      You can try out the different prop values in the \`knobs\` section below!
      
      \`Usage:\`
      ~~~js
        <Avatar 
          size='small'
          name='John Doe'
          src='/src/to/avatar' 
        />
      ~~~
    `,
    () => (
      <div>
        <Avatar
          name={text('name',contributor.name)}
          src={text('src',contributor.src)}
          size={select('size', sizeOptions ,'medium')}
        />
      </div>
    ),
    {
      inline: true,
      propTables: false,
    }
  );
