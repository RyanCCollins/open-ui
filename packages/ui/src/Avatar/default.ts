// tslint:disable
export default 'https://cloud.githubusercontent.com/assets/13810084/23538009/f9e348ae-ff9d-11e6-9f7a-485e8247a026.png';