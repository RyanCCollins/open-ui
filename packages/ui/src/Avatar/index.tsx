import * as React from 'react';
import { Props } from './types';
import Component from './styles';

export default function Avatar({
  ...props,
}: Props): JSX.Element {
  return (
    <Component {...props} />
  );
};
