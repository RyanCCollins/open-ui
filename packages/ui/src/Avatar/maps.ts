export default {
  thumb: 50,
  small: 100,
  medium: 150,
  large: 250,
};
