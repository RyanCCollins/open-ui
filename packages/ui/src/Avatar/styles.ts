import remStringFromPx from '../utils';
import map from './maps';
import { Props } from './types';
import defaultUrl from './default';
const styled = require('styled-components').default;
const { css } = require('styled-components');

const propsToSize = ({ size }: Props) => {
  const imageSize = size || 'medium';
  const px = map[imageSize];
  const rem = remStringFromPx(px);
  return css`
    width: ${rem};
    min-height: ${rem};
  `;
};

export default styled.div`
  background: url(${({ src }) => src || defaultUrl});
  background-size: cover;
  background-position: center;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
  padding: 4px;
  line-height: 1.42857143;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 50%;
  ${propsToSize}
`;
