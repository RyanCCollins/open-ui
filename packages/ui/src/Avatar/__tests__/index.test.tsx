import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Avatar from '../';
import props from './__mocks__/avatarMocks.mock';

describe('<Avatar {...props} />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Avatar {...props} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with thumb size', () => {
    const wrapper = shallow(
      <Avatar {...props} size="thumb" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with small size', () => {
    const wrapper = shallow(
      <Avatar {...props} size="small" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with medium size', () => {
    const wrapper = shallow(
      <Avatar {...props} size="medium" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with large size', () => {
    const wrapper = shallow(
      <Avatar {...props} size="large" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
