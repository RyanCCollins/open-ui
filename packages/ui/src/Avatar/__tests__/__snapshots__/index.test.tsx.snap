// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar {...props} /> should render with default props 1`] = `
<styled.div
  name="Abhishek Ghosh"
  src="https://avatars3.githubusercontent.com/u/19292575"
/>
`;

exports[`<Avatar {...props} /> should render with large size 1`] = `
<styled.div
  name="Abhishek Ghosh"
  size="large"
  src="https://avatars3.githubusercontent.com/u/19292575"
/>
`;

exports[`<Avatar {...props} /> should render with medium size 1`] = `
<styled.div
  name="Abhishek Ghosh"
  size="medium"
  src="https://avatars3.githubusercontent.com/u/19292575"
/>
`;

exports[`<Avatar {...props} /> should render with small size 1`] = `
<styled.div
  name="Abhishek Ghosh"
  size="small"
  src="https://avatars3.githubusercontent.com/u/19292575"
/>
`;

exports[`<Avatar {...props} /> should render with thumb size 1`] = `
<styled.div
  name="Abhishek Ghosh"
  size="thumb"
  src="https://avatars3.githubusercontent.com/u/19292575"
/>
`;
